### PR TITLE
Feature/issue 373 improve testing for jupiter

### PR DESCRIPTION
**Pull request by l-1squared**

### CODE REVIEW

File: jgiven-tests/src/main/java/com/tngtech/jgiven/tests/GuaranteedFieldRealTest.java [hunk 1]
```diff
@@ -5,17 +5,22 @@
 import com.tngtech.jgiven.annotation.ProvidedScenarioState;
 import com.tngtech.jgiven.junit.ScenarioTest;
 import org.junit.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
 
-public class GuaranteedFieldRealTest extends ScenarioTest<GuaranteedFieldRealTest.RealGiven, GuaranteedFieldRealTest.RealWhen, GuaranteedFieldRealTest.RealThen> {
+@ExtendWith(JGivenReportExtractingExtension.class)
```
Comment (l-1squared) [inline jgiven-tests/src/main/java/com/tngtech/jgiven/tests/GuaranteedFieldRealTest.java:10]: probably not necessary, we don't test these with JUnit5

Comment (l-1squared) [reply jgiven-tests/src/main/java/com/tngtech/jgiven/tests/GuaranteedFieldRealTest.java:10]: also, we should probably move those into core at some point

File: jgiven-tests/src/test/java/com/tngtech/jgiven/junit5/Junit5TestExecutionResult.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    @Override
+    public int getFailureCount() {
+        return (int) result.values().stream()
+            .filter(result -> result.getStatus() == org.junit.platform.engine.TestExecutionResult.Status.FAILED)
+            .count();
+    }
+
+    @Override
+    public String getFailureMessage(int i) {
```
Comment (l-1squared) [inline jgiven-tests/src/test/java/com/tngtech/jgiven/junit5/Junit5TestExecutionResult.java:28]: Explain better how these failure message routine came to be

File: jgiven-tests/src/test/java/com/tngtech/jgiven/testframework/TestFramework.java [hunk 3]
```diff
@@ -2,5 +2,6 @@
 
 public enum TestFramework {
     JUnit,
+    JUnit5,
     TestNG
 }
```
Comment (l-1squared) [inline jgiven-tests/src/test/java/com/tngtech/jgiven/testframework/TestFramework.java:7]: fix

File: jgiven-tests/src/test/java/com/tngtech/jgiven/testframework/TestFrameworkExecutionTest.java [hunk 4]
```diff
@@ -1,36 +1,37 @@
...
 
     public final TestFramework testFramework;
 
     @Parameters
     public static Iterable<Object[]> testFrameworks() {
-        return Arrays.asList( new Object[][] {
-            { TestFramework.JUnit },
-            { TestFramework.TestNG },
-        } );
+        return Arrays.stream(TestFramework.values())
```
Comment (l-1squared) [inline jgiven-tests/src/test/java/com/tngtech/jgiven/testframework/TestFrameworkExecutionTest.java:29]: try if TestFramework.values() will work just fine.

Comment (l-1squared) [reply jgiven-tests/src/test/java/com/tngtech/jgiven/testframework/TestFrameworkExecutionTest.java:29]: Documentation indicates that this won't be the case

File: jgiven-tests/src/main/java/com/tngtech/jgiven/tests/JUnit5AfterMethodTests.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    @Test
+    public void a_failing_JUnit_5_test() {
+        given().nothing();
+        when().a_step_fails();
+        then().something_happened();
+    }
+
+    @Test
+    public void a_succeding_JUnit5_test() {
```
Comment (fudler) [inline jgiven-tests/src/main/java/com/tngtech/jgiven/tests/JUnit5AfterMethodTests.java:29]: Typo in succeeding

File: jgiven-tests/src/test/java/com/tngtech/jgiven/junit5/JUnit5ExecutorTest.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    @Test
+    public void tests_with_scenario_modifications_in_after_method() {
+        given().junit5_tests_with_scenario_modifications_in_after_method();
+        when().the_test_class_is_executed_with_JUnit5();
+        then().each_scenario_contains_$_cases(2);
+    }
+
+    @Test
+    @Issue("#25")
+    public void test_classes_with_only_ignored_test_result_in_a_valid_report() {
```
Comment (fudler) [inline jgiven-tests/src/test/java/com/tngtech/jgiven/junit5/JUnit5ExecutorTest.java:25]: Is it somehow possible to give this method some better name? Needed some time to figure out which word groups belong to the same context. Like "test_classes_which_consist_of_only_ignored_tests" or something.

**Review by fudler — APPROVED** [review]

**Pull request merged**